app/controllers/user_badges_controller.rb

```diff
@@ -91,6 +94,20 @@ def destroy
     render json: success_json
   end
 
+  def favorite
+    params.require(:id)
+    user_badge = UserBadge.find(params[:id])
+    user_badges = user_badge.user.user_badges
+
+    if !user_badge.is_favorite && user_badges.where(is_favorite: true).count >= MAX_FAVORITES
+      render json: failed_json, status: 403
+    else
+      user_badge.update_attribute(:is_favorite, !user_badge.is_favorite)
```

Comment:
You could use `toggle` which is shorter

```suggestion
      user_badge.toggle!(:is_favorite)
```

Reply:
Ah, years without using rails and I forgot how delightful it was! Thanks for pointing it out :)